utils_test.py

```diff
@@ -264,6 +264,89 @@ def test_string_to_datetime_conversion_returns_correct_datetime(self):
             utils.convert_string_to_naive_datetime_object(time_string),
             initial_time)
 
+    def test_create_string_from_largest_unit_in_timedelta_returns_days(self):
+        timedelta_object = datetime.timedelta(
+            days=4, hours=1, minutes=1, seconds=1)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '4 days')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_a_day(self):
+        timedelta_object = datetime.timedelta(
+            days=1, hours=1, minutes=1, seconds=1)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '1 day')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_hours(self):
+        timedelta_object = datetime.timedelta(
+            days=0, hours=2, minutes=1, seconds=1)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '2 hours')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_a_hour(self):
+        timedelta_object = datetime.timedelta(
+            days=0, hours=1, minutes=1, seconds=1)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '1 hour')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_minutes(self):
+        timedelta_object = datetime.timedelta(
+            days=0, hours=0, minutes=4, seconds=1)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '4 minutes')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_a_minute(
+            self):
+        timedelta_object = datetime.timedelta(
+            days=0, hours=0, minutes=1, seconds=12)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '1 minute')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_minute_if_sec(
+            self):
+        timedelta_object = datetime.timedelta(
+            days=0, hours=0, minutes=0, seconds=1)
+
+        time_string = (
+            utils.create_string_from_largest_unit_in_timedelta(timedelta_object)
+        )
+
+        self.assertEqual(time_string, '1 minute')
+
+    def test_create_string_from_largest_unit_in_timedelta_returns_a_min_if_msec(
```

Comment:
Although it's conflated, I suggest also having tests for the 0 and negative timedelta situations. These should have defined output, or at least errors.